### PR TITLE
Exclude armv7 arch from iOS add-to-app integration test hosts

### PR DESCRIPTION
**Pull request by jmagman**
To support removing `armv7` iOS arch from the Flutter.framework (https://github.com/flutter/engine/pull/32664), the add-to-app iOS host apps must exclude building that arch.

All existing add-to-apps hosts will need to do this migration manually as well.  I will update the samples, and the website with instructions:
Screen Shot 2022-04-15 at 2 08 37 PM

See also #101943

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by jmagman** [comment]
This doesn't have tests, but it is exercised by `module_test_ios` among others, which I expect to break when https://github.com/flutter/engine/pull/32664 rolls unless this is changed.

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 1]
```diff
@@ -439,10 +439,15 @@ end
...
+        final String builtHostBinary = path.join(
           archivedAppPath,
           'Host',
-        ));
+        );
+        checkFileExists(builtHostBinary);
+        if ((await fileType(builtHostBinary)).contains('armv7')) {
+          throw TaskResult.failure('Unexpected armv7 architecture slice in $builtHostBinary');
+        }
+        await containsBitcode(builtHostBinary);
```
Comment (christopherfujino) [inline dev/devicelab/bin/tasks/module_test_ios.dart:450]: what does this do?

Comment (jmagman) [reply dev/devicelab/bin/tasks/module_test_ios.dart:450]: You're right, this is a bug.  Will fix all the spots in this file.

**Comment by jmagman** [comment]
Hmm, the module tests passed in https://github.com/flutter/flutter/pull/102012 (https://github.com/flutter/engine/pull/32664) without this change.  I'm going to close this and think more about why this wasn't needed. 🤔
Will follow up in another PR to check the `containsBitcode` result.

**Comment by jmagman** [comment]
> Will follow up in another PR to check the `containsBitcode` result.

https://github.com/flutter/flutter/pull/102016

**Pull request closed without merging**